src/main/java/io/github/bobocodebreskul/context/registry/AnnotatedBeanDefinitionReader.java

```diff
@@ -98,6 +112,47 @@ public final BeanDefinitionRegistry getBeanDefinitionRegistry() {
     return beanDefinitionRegistry;
   }
 
+  private Optional<String> extractBeanName(Class<?> beanClass) {
+    Set<String> componentAnnotations = Arrays.stream(beanClass.getAnnotations())
+      .filter(this::isItComponentAnnotation)
+      .map(annotation -> getClassAnnotationValue(beanClass, annotation.annotationType()))
+      .filter(beanName -> nonNull(beanName) && !beanName.isEmpty())
+      .collect(Collectors.toSet());
+    if (componentAnnotations.isEmpty()) {
+      return Optional.empty();
+    } else if (componentAnnotations.size() == 1) {
+      return componentAnnotations.stream().findFirst();
+    } else {
+      String beanNames = String.join(", ", componentAnnotations);
+      log.error(
+        "For bean {} was found several different names definitions: [{}]. Please choose one.",
+        beanClass.getName(), beanNames);
+      throw new IllegalStateException(
+        UNCERTAIN_BEAN_NAME_EXCEPTION_MSG.formatted(beanClass.getName(), beanNames));
+    }
+  }
+
+
+  private String getClassAnnotationValue(Class<?> classType,
+    Class<? extends Annotation> annotationType) {
+    String value = null;
+    Annotation annotation = classType.getAnnotation(annotationType);
+    if (annotation != null) {
+      try {
+        value = (String) annotation.annotationType().getMethod(COMPONENT_NAME_FIELD)
+          .invoke(annotation);
+      } catch (Exception ignore) {
+      }
+    }
+    return value;
+  }
+
+  private boolean isItComponentAnnotation(Annotation annotation) {
```

Comment:
'isComponentAnnotation' would be more clear and concise.
Additionally, I'd think about to extract this method to some utility class, since this logic may be needed for some other classes